### PR TITLE
Add tests to verify login

### LINKED ISSUE #3: Add Playwright Tests for Login Functionality
## Issue Title:
**Add Playwright Tests for Login Functionality**

## Issue Description:
**Objective:** Automate the testing of the login functionality in neetoPlanner using Playwright. This test will also serve as the global setup for subsequent tests.

### Test Case Steps:
1. **Visit Subdomain:** Navigate to the specific neetoPlanner subdomain.
2. **Input Email:** Enter the user's email address in the email input field.
3. **Input OTP:** Enter a random six-digit OTP for authentication.
4. **Confirm Login:** Click the login button to complete the authentication process.

### Requirements:
- Ensure the test navigates to the correct URL.
- Validate input fields for email and OTP.
- Check the login process to ensure it handles the OTP correctly.
- Use environment variables for sensitive data like the subdomain URL and user credentials.

### Expected Result:
The user should be successfully logged in to the dashboard upon entering valid credentials and OTP.

This issue will guide the development of the first test script and ensure that the basic authentication mechanism is robustly tested.

### PR DESCRIPTION
**Pull request by Srutip04**
Closes #3

### CODE REVIEW

File: constants/selectors/login.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export const LOGIN_SELECTORS = {
+    emailInput: "neeto-auth-email-input-field",
+    loginButton: "neeto-auth-login-button",
+    loginCodeInput: "Enter 6 digit login code",
```
Comment (shreyakurian02) [inline constants/selectors/login.ts]: This isn't a selector. Create another folder called texts within constants and store this there.
`constants/texts/****.ts`

File: e2e/fixtures/index.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// fixtures/index.ts
```
Comment (shreyakurian02) [inline e2e/fixtures/index.ts]: There isn't any need to specify file name at the top of every file. It is quite evident from the folder structure.
Remove unwanted comments. Too many comments can reduce readability.

File: e2e/fixtures/index.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import LoginPage from "../poms/login";
+
+// Defining an interface for extended fixtures to specify additional fixtures
+interface ExtendedFixtures {
+  loginPage: LoginPage;
+}
+
+// Exporting a custom test fixture named 'test' using the 'extend' function
+export const test = base.extend<ExtendedFixtures>({
+  // Defining the 'loginPage' fixture
```
Comment (shreyakurian02) [inline e2e/fixtures/index.ts]: Remove unwanted comments.

File: e2e/poms/login.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  }: {
+    email: string;
+    loginCode: string;
+  }): Promise<void> => {
+    await this.page.getByTestId(LOGIN_SELECTORS.emailInput).fill(email);
+    await this.page.getByTestId(LOGIN_SELECTORS.loginButton).click();
+    await this.page.getByPlaceholder(LOGIN_SELECTORS.loginCodeInput).fill(loginCode);
+    //verify successful login
+    await expect(this.page.getByTestId(LOGIN_SELECTORS.mainHeader)).toBeVisible({timeout: 50000})
+    // await this.page.waitForURL('https://sruti-pingua-iiit-bbsr.neetoplanner.net/dashboard/active');
```
Comment (shreyakurian02) [inline e2e/poms/login.ts]: Remove commented code.

File: e2e/tests/login.setup.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+// Describing the test suite for the Login page
+test.describe("Login page", () => {
+  let loginCode : string;
+
+  test.beforeEach(()=>{
+    loginCode = faker.string.numeric({length:testData.loginCodeLength})
+})
+
+  test("should login with the correct credentials", async ({ page ,loginPage}) => {
```
Comment (shreyakurian02) [inline e2e/tests/login.setup.ts]: Added appropriate step blocks and remove comments.
https://courses.bigbinaryacademy.com/learn-qa-automation-using-playwright/best-practices/#bigbinary-best-practices

File: playwright.config.ts [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    // },
+    // {
+    //   name: 'Google Chrome',
+    //   use: { ...devices['Desktop Chrome'], channel: 'chrome' },
+    // },
+  ],
+
+  /* Run your local dev server before starting the tests */
+  // webServer: {
+  //   command: 'npm run start',
```
Comment (shreyakurian02) [inline playwright.config.ts]: Remove unwanted code.

File: constants/utils/testData.ts [hunk 7]
```diff
@@ -0,0 +1,4 @@
+export const testData = {
+    testEmail: "cpts9gnqty9-planner-sruti_pingua-iiit_bbsr@bigbinary.com",
+    loginCodeLength: 6,
+};
```
Comment (shreyakurian02) [inline constants/utils/testData.ts]: This is a constant and not util. Also ensure that screaming snake case is used for defining constants.

**Review by shreyakurian02 — CHANGES_REQUESTED** [review]

**Comment by Srutip04** [comment]
I've addressed the requested changes in this PR.Please review the changes.

**Review by shreyakurian02 — APPROVED** [review]

**Pull request merged**